DataStructure/Stack/Stackhomework.php

```diff
@@ -0,0 +1,83 @@
+<?php
+class Stack {
+    /** @var array stack element */
+    private $elements;
+
+    public function __construct()
+    {
+        $this->elements = array(); //initialize stack element
+    }
+
+    /**
+    * insert an element
+    * @param string num
+    */
+    public function push($ele)
+    {
+        $this->elements[] = $ele;
+    }
+
+    /**
+    * delete top element
+    */
+    public function pop()
+    {
+        if (!$this->isEmpty()) { //check if stack is not empty
+            unset($this->elements[sizeof($this->elements) - 1]); // unset use to destroy a variable (https://www.php.net/manual/en/function.unset)
```

Comment:
When you use `unset()` an element, the element will be dropped from the array, it's mean your array will be lack an index, which is dropped by `unset()`. As you can see in the picture below:

![image](https://user-images.githubusercontent.com/83938068/118077237-ea00d700-b3dd-11eb-9032-19176953bdfe.png)
 _Read more_: https://www.php.net/manual/en/function.unset.php
 
 Please find a better way to pull out an element without changing the order of the array.